types/utxo.go

```diff
@@ -0,0 +1,142 @@
+package types //might need to be adjusted
+
+import (
+	crypto "github.com/tendermint/go-crypto"
+	"errors"
+	//sdk "github.com/cosmos/cosmos-sdk/types"
+
+)
+
+// UTXO is a standard unspent transaction output
+// Has pubkey for authentication
+type UTXO interface {
+	// Decided to use address instead of public keys
+	// since each utxo newly created will be created from an
+	// Ethereum address
+	GetAddress() crypto.Address
+	SetAddress(crypto.Address) error // errors if already set
+
+	//Get and Set denomination of the utxo. Is uint64 appropriate type?
+	GetDenom() uint64
+	SetDenom(uint64) error //errors if already set
+
+	Get(key interface{}) (value interface{}, err error)
+	Set(key interface{}, value interface{}) error
+	
+	//TODO: ADD SUPPORT FOR DIFFERENT COINS
+
+}
+
+type UTXOHolder interface {
```

Comment:
When I was going through the code, I understood that `UTXOHolder` and `UTXOKeeper` are two different things, but at least for me, their names were confusing and could be confusing for others down the line as well. (kinda hard to differentiate between a 'holder' and 'keeper' if you're going through code quickly)

It seems you modeled `UTXOKeeper` after `CoinKeeper` - that makes sense. In that case, what do you think about renaming `UTXOHolder` to something else?